### PR TITLE
Bug 1711439: Adds readiness probe to DaemonSet asset

### PR DESCRIPTION
**Pull request by danehans**
This PR adds `readinessProbe` to the DNS DaemonSet. The values for all fields accept `httpGet`, are from the kube-apiserver and kube-scheduler pods:

- `initialDelaySeconds`: Number of seconds after the container has started before probes are initiated. Defaults to 0 (no delay).
- `periodSeconds`: How often (in seconds) to perform the probe. Default to 10 seconds. Minimum value is 1.
- `successThreshold `: Minimum consecutive successes for the probe to be considered successful after having failed. Defaults to 1. Minimum value is 1.
- `failureThreshold `: Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to 3. Minimum value is 1.
- `timeoutSeconds`: Number of seconds after which the probe times out. Defaults to 1 second. Minimum value is 1.

### CODE REVIEW

**Comment by Miciah** [comment]
/lgtm
/hold
Holding in case Dan wants to review the parameters (they look reasonable enough to me, but I don't have a great feel for how they should be set).

File: assets/dns/daemonset.yaml [hunk 1]
```diff
@@ -27,6 +27,16 @@ spec:
         - containerPort: 9153
           name: metrics
           protocol: TCP
+        readinessProbe:
+          httpGet:
+            path: /health
+            port: 8080
+            scheme: HTTP
+          initialDelaySeconds: 10
```
Comment (ironcladlou) [inline assets/dns/daemonset.yaml]: Might be a little conservative — normal startup of the coredns process is sub-second. Not sure whether we should be shaving seconds here.

**Comment by ironcladlou** [comment]
Thanks!

/hold cancel

**Comment by ironcladlou** [comment]
@danehans please investigate https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-dns-operator/111/pull-ci-openshift-cluster-dns-operator-master-e2e-aws-upgrade/46?

/retest

**Comment by Miciah** [comment]
> please investigate https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-dns-operator/111/pull-ci-openshift-cluster-dns-operator-master-e2e-aws-upgrade/46?

One of the failures looks like https://bugzilla.redhat.com/show_bug.cgi?id=1690087.

**Pull request merged**